WcaOnRails/app/models/competition.rb

```diff
@@ -427,6 +445,10 @@ def dangerously_close_to?(c)
     self.kilometers_to(c) <= NEARBY_DISTANCE_KM_DANGER && days_until.abs < NEARBY_DAYS_DANGER
   end
 
+  def results_uploaded?
```

Comment:
Tim's idea didn't solve it, unfortunately :(
Querying for results for 2k+ competitions runs very slow, so I just removed it for now.